api/v1alpha1/clusterconfig_types.go

```diff
@@ -151,7 +151,7 @@ type NutanixClusterConfigSpec struct {
 	Addons *NutanixAddons `json:"addons,omitempty"`
 
 	// +kubebuilder:validation:Optional
-	ControlPlane *NutanixNodeConfigSpec `json:"controlPlane,omitempty"`
+	ControlPlane *NutanixControlPlaneSpec `json:"controlPlane,omitempty"`
```

Comment:
how do we handle existing cluster's upgrade to this schema? should we upgrade api version?

Reply:
As I explained in the pr description, this does not affect the cluster class variable schema so will only affect downstream projects using this in go code, of which I can only think of one potential which is konvoy. I do not think we need to bump an API version for this, even more so that it's still alpha versioned.